daemon/exec.go

```diff
@@ -164,18 +164,23 @@ func (d *Daemon) ContainerExecStart(ctx context.Context, name string, stdin io.R
 		return fmt.Errorf("Error: Exec command %s is already running", ec.ID)
 	}
 	ec.Running = true
+	ec.Unlock()
+
+	c := d.containers.Get(ec.ContainerID)
+	logrus.Debugf("starting exec command %s in container %s", ec.ID, c.ID)
+	d.LogContainerEvent(c, "exec_start: "+ec.Entrypoint+" "+strings.Join(ec.Args, " "))
+
 	defer func() {
 		if err != nil {
 			ec.Running = false
```

Comment:
Only if changes to ec.Running and ec.ExitCode need to be atomic. There were no lock protection in containerd callback - so I added no lock here. Should I  add the lock? Add the lock in containerd callback as well?